packages/block-editor/src/components/block-settings-menu/index.js

```diff
@@ -70,7 +70,10 @@ export function BlockSettingsMenu( { clientIds, onSelect } ) {
 							{ ! isLocked && canDuplicate && (
 								<MenuItem
 									className="editor-block-settings-menu__control block-editor-block-settings-menu__control"
-									onClick={ onDuplicate }
+									onClick={ () => {
+										onClose();
+										onDuplicate();
+									} }
```

Comment:
maybe just use lodash flow here and elsewhere?  eg `flow( onClose, onDuplicate )`

Reply:
Also, using this pattern for onClick callbacks means every time this component is rendered the `onClick` prop value changes.  So it might be better to avoid doing that (I realize this pattern is used elsewhere in this component).

Reply:
I'll rewrite to use `flow()` 👍 

> So it might be better to avoid doing that

What would be the preferred approach in such a case then?

Reply:
> What would be the preferred approach in such a case then?

I missed this was a functional component so should be okay.